### PR TITLE
Use mathjax context in demos

### PR DESCRIPTION
**Pull request by devpow112**
Also update documentation. Should have done this with my other PR but didn't. This expands on #2288 and switches `core` demos to use the new file. Required a bit of re-work since didn't realize we had a demo page that disabled it. This should work, you need to explicitly set `latex=false` on a URL for a demo page or test to disable the test context render latex property, otherwise it's enabled.

### CODE REVIEW

File: tools/mathjax-test-context.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
-document.documentElement.dataset.mathjaxContext = JSON.stringify({
+const disabled = window.location.search.indexOf('latex=false') !== -1;
+
+document.getElementsByTagName('html')[0].dataset.mathjaxContext = JSON.stringify({
 	outputScale: 1.1,
-	renderLatex: true
+	renderLatex: !disabled
```
Comment (devpow112) [inline tools/mathjax-test-context.js:5]: Will be `true` unless explicitly disabled. This seems like a better state for most tests and demo pages using HTML block.

File: components/html-block/README.md [hunk 2]
```diff
@@ -99,29 +100,41 @@ Examples are provided to display how user-authored math can be embedded within y
...
 <script type="module">
   import '@brightspace-ui/core/components/html-block/html-block.js';
-
-  window.D2L = {};
-  D2L.LP = {};
-  D2L.LP.Web = {};
-  D2L.LP.Web.UI = {};
-  D2L.LP.Web.UI.Flags = {
-    Flag: (feature, defaultValue) => {
-      if (feature === 'us125413-mathjax-render-latex') return true;
```
Comment (devpow112) [inline components/html-block/README.md:115]: The `us125413-mathjax-render-latex` flag isn't used anywhere and isn't required for LaTeX to work from what I can tell.

Comment (devpow112) [reply components/html-block/README.md:115]: Was marked as removed as part of https://github.com/Brightspace/lms-launch-darkly-flags/pull/2925 and later deleted

**Comment by devpow112** [comment]
> > This looks ok to me.
> > @grant-cleary : do you have any concerns with this?
> 
> No real concerns here. I'm hoping folks with components integrated into the LMS don't start overriding these context objects of their own accord, but... Well there's nothing we can really do to stop that, so 🤷.

Yeah this is definitely not intended for that, just for testing/demo help

**Comment by grant-cleary** [comment]
> Yeah this is definitely not intended for that, just for testing/demo help

Oh yeah, sorry, not specifically your change I'm thinking about here. I'm just not sure I like how we decided to talk about the MathJax context in the readme. The wording almost makes it sound like this is a thing we expect consumers to do manually, but... Well we kind of do for off-stack services, but not for stuff integrated in the LMS.

... Anyways, just idle musings. I'm happy with this as is. :P

**Comment by devpow112** [comment]
> > No real concerns here. I'm hoping folks with components integrated into the LMS don't start overriding these context objects of their own accord, but... Well there's nothing we can really do to stop that, so 🤷.
> 
> Maybe we should include a console.warn/error in the `mathjax-test-context.js` file?

I could add something like `console.warn('Using mathjax test context, this is intended for demo pages and tests only')` if that woks for you @dbatiste @grant-cleary.

**Comment by grant-cleary** [comment]
> I could add something like `console.warn('Using mathjax test context, this is intended for demo pages and tests only')` if that woks for you @dbatiste @grant-cleary.

Seems good to me.

**Pull request merged**